src/components/SendTransaction/SendTransactionFlow.tsx

```diff
@@ -73,14 +77,18 @@ export const SendTransactionFlow = ({ onCancel }: { onCancel: () => void }) => {
           onRestartFlow={() => {
             setFormData(initialFormData);
             setCurrentStage(SendState.CREATE);
+            dispatch(resetSendTxAction());
           }}
           onCancel={onCancel}
         />
       )}
 
       {currentStage === SendState.ERROR && (
         <FailedTransaction
-          onEditTransaction={() => setCurrentStage(SendState.CREATE)}
+          onEditTransaction={() => {
+            setCurrentStage(SendState.CREATE);
+            dispatch(resetSendTxAction());
```

Comment:
do we want to reset here? If they're editing they should be able to edit their old inputs no?